.github/workflows/ci-test.yml

```diff
@@ -25,6 +25,8 @@ jobs:
           # On `push` this value will be empty and will "do-the-right-thing"
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0 #get all the history!!!
+      - name: Git Hooks and Checks
+        run: ./scripts/git-checks.sh
```

Comment:
this needs to run after the build to catch when people dont check in the Cargo.lock changes.   That what this check most often finds.

Reply:
huh not the script I thought it was....

Reply:
This does a number of things like making sure that filenames are not bad/malicious and that there are no merge commits etc.